pkg/cloudprovider/aws/fleet/packing/packing.go

```diff
@@ -45,12 +47,108 @@ func NewPacker(ec2 ec2iface.EC2API) *PodPacker {
 // Pack returns the packings for the provided pods. Computes a set of viable
 // instance types for each packing of pods. Instance variety enables EC2 to make
 // better cost and availability decisions.
+// Pods provided are all schedulable in the same zone as tightly as possible.
 func (p *PodPacker) Pack(ctx context.Context, pods []*v1.Pod) ([]*Packings, error) {
 	zap.S().Debugf("Successfully packed %d pods onto %d nodes", len(pods), 1)
-	return []*Packings{
-		{
-			InstanceTypeOptions: []string{"m5.large"}, // TODO, prioritize possible instance types
-			Pods:                pods,
-		},
-	}, nil
+
+	// 1. Arrange pods in decreasing order by the amount of CPU requested, if
+	// CPU requested is equal compare memory requested.
+	sort.Sort(sort.Reverse(byResourceRequested{pods}))
```

Comment:
Why do we need to reverse if we control the comparator byResourceRequested. Can't our comparator just do the right thing?

Reply:
sort.Sort implements a Less(a, b) to compare two objects expects true `if a < b` and it might be confusing to return the reverse.

Reply:
So, the way it works is `sort.Sort` calls the `Less(i, j int)` method to compare the two values.
In this case, `byResourceRequested` implements a `Less` method (comparator) and returns the right thing, when value at index i < index j, return true. This will work if you are just doing an increasing sort.
Here, we need to reverse, so `sort.Reverse` reverses the decision made by `Less(i, j)` implemented by `byResourceRequested`. 

Also, we could have reversed this decision in the `Less` method by returning false when index i < index j. With this we would only need to do `sort.Sort(byResourceRequested{pods})`. This can be very confusing because Less is doing the reverse of what it's suppose to.

Reply:
Ah, sorry - I was making an incorrect assumption about what `sort.Reverse` does!